src/plugins/workspace/public/components/delete_workspace_modal/delete_workspace_modal.tsx

```diff
@@ -16,16 +16,65 @@ import {
   EuiSpacer,
   EuiText,
 } from '@elastic/eui';
+import { WorkspaceAttribute } from 'opensearch-dashboards/public';
+import { i18n } from '@osd/i18n';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { WorkspaceClient } from '../../workspace_client';
 
 interface DeleteWorkspaceModalProps {
-  selectedItems: string[];
   onClose: () => void;
-  onConfirm: () => void;
+  selectedWorkspace?: WorkspaceAttribute | null;
+  ifNavigate: boolean;
```

Comment:
what does `ifNavigate` used for? Only workspace list page have delete functionality, maybe we can remove this property as don't do direct after deleting.

Reply:
Workspace creator page will also use delete workspace model. And it will navigate after deleting in workspace creator page.

Reply:
How about renaming to `shouldNavigate`?  `ifNavigate` seems not a good naming.

Reply:
maybe `returnToHome` to make it clear where will it redirect to?